tests/ui/abi/c-zst.rs

```diff
@@ -1,27 +1,40 @@
-//@ revisions: other other-linux x86_64-pc-windows-gnu s390x-linux sparc64-linux powerpc-linux
 //@ normalize-stderr-test: "(abi|pref|unadjusted_abi_align): Align\([1-8] bytes\)" -> "$1: $$SOME_ALIGN"
+
+// Ignore the ZST revisions
```

Comment:
I don't understand the comment. What are we supposed to ignore?

Reply:
The old code had two sections,"ignore the ZST" and "pass the ZST indirectly". That structure seems to have been lost and now it's just confusing.

Reply:
I found the previous structure confusing too, tbh.

Reply:
@RalfJung I don't feel confident that our handling of passing the ZST by indirection is correct for the ABIs in question.

Reply:
> I found the previous structure confusing too, tbh.

There are two kinds of targets: those where ZST are ignored, and those where they are passed indirectly (which may seem nonsensical but it's what it is). It is helpful to group the revisions for the test by which category a target falls into.

Reply:
@RalfJung I did a bit more studying to satisfy my curiosity a bit more as to exactly _why_ so many ABIs decide ZSTs consume space, and organized the revisions accordingly again, plus added some remarks as to the history.

Reply:
Thanks!

I was going to suggest a better place for the comment would be the code in rustc that implements this ABI logic. That used to be one central place but https://github.com/rust-lang/rust/commit/b1493ba5194fcc5cfe4f6315db288e4e18509110 recently moved it to the corresponding targets... which is probably a good call but makes it harder to to put a good comment in a central place.

Reply:
Yeah, "the code may or may not be in one place" is basically the reason I've been leaving these sorts of comments by the test cases. Tests in many ways describe the spec that we're adhering to, but the implementation can be whatever... including "spread out across 100 modules".

I may find a better home for some of this text at some point, but not today.